### PR TITLE
Adaptable gameplay speed

### PR DESCRIPTION
**Pull request by LuukSantegoeds**
# Description
Change the delta time modifier with page up and page down.

**Please note!**
Changing the modifier has no effect on horizontal movement because of terminal velocity!

## Screenshot(s)
No

# How can this be tested?
Combine with [this PR](https://github.com/Brick-Studios/BrickEngine/pull/49). Start the game and change the modifier with page up and page down.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Included 1 or more screenshots
- [X] Code is const correct

### CODE REVIEW

File: lib/systems/game_speed_system.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#include "player_input.hpp"
+
+GameSpeedSystem::GameSpeedSystem(std::shared_ptr<EntityManager> em, double *delta_time_modifier) : System(em), delta_time_modifier(delta_time_modifier) {};
+
+void GameSpeedSystem::update(double delta_time) {
+    std::ignore = delta_time;
+
+    auto input = BrickInput<PlayerInput>::getInstance();
+    double step_size = 0.2;
```
Comment (Bram-Boris) [inline lib/systems/game_speed_system.cpp]: This can be a const in hpp

File: lib/systems/game_speed_system.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#include "systems/game_speed_system.hpp"
+
+#include <utility>
+
+#include "brickengine/input.hpp"
+
+#include "player_input.hpp"
+
+GameSpeedSystem::GameSpeedSystem(std::shared_ptr<EntityManager> em, double *delta_time_modifier) : System(em), delta_time_modifier(delta_time_modifier) {};
```
Comment (Bram-Boris) [inline lib/systems/game_speed_system.cpp]: I think passing by reference would be better.

**Review by Bram-Boris — CHANGES_REQUESTED** [review]
I think this feature can be in the engine.

File: lib/systems/game_speed_system.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#include "systems/game_speed_system.hpp"
+
+#include <utility>
+
+#include "brickengine/input.hpp"
+
+#include "player_input.hpp"
```
Comment (JanSchollaert) [inline lib/systems/game_speed_system.cpp:7]: Unnecessary whitelines but not that problematic (should be comment not approve)

**Review by JanSchollaert — APPROVED** [review]

**Comment by JanSchollaert** [comment]
@Bram-Boris I thought it should be in the engine as well but wasn't sure because you are using inputs to set them and I wasn't sure if thats game specific

**Comment by LuukSantegoeds** [comment]
I think that having the modifier system in the game is perfectly fine. Mark agrees with me so the truth has been spoken :P

**Review by JanSchollaert — DISMISSED** [review]
nice

**Comment by Peter-JanGootzen** [comment]
We could maybe fix the vertical velocity by adding a separate speed_modifier parameter to System::update.

File: include/player_input.hpp [hunk 4]
```diff
@@ -7,7 +7,10 @@ enum PlayerInput {
     SHOOT,
     GRAB,
     MOUSE_LEFT,
-    MOUSE_RIGHT
+    MOUSE_RIGHT,
+    PAGE_DOWN,
+    PAGE_UP,
+    HOME
```
Comment (JanSchollaert) [inline include/player_input.hpp]: Change these to SPEED_UP SPEED_DOWN AND SPEED_RESET, or something like this

**Review by JanSchollaert — APPROVED** [review]
nice

**Review by Bram-Boris — APPROVED** [review]

**Pull request merged**